src/uwtools/tests/test_rocoto.py

```diff
@@ -129,6 +129,20 @@ def test__add_task_dependency_fail(self, instance, root):
         with raises(UWConfigError):
             instance._add_task_dependency(e=root, config=config)
 
+    def test__add_data_dependency(self, instance, root):
+        config = {"datadep": {"age": "120"}}
+        instance._add_task_dependency_operand(e=root, config=config)
+        datadep = root[0]
+        assert datadep.tag == "datadep"
+        assert datadep.get("age") == "120"
+
+    def test__add_time_dependency(self, instance, root):
+        config = {"timedep": {"offset": "&DEADLINE;"}}
+        instance._add_task_dependency_operand(e=root, config=config)
+        timedep = root[0]
+        assert timedep.tag == "timedep"
+        assert timedep.get("offset") == "&DEADLINE;"
+
```

Comment:
I split `taskdep`, `datadep` and `timedep` up into their own test cases. 

I can merge these under a `test__add_operand_dependency` if that is more ideal. 

Currently `test__add_task_dependency`, `test__add_data_dependency` and `test__add_time_dependency` each call `_add_task_dependency_operand` individually and leads to quite a bit of duplication.

Reply:
I like the granular tests, one per dependency type. If the three tests were more than a few lines each and shared a lot in common, I'd maybe lean toward DRY and try to use `@pytest.mark.parametrize` to fold them into a single test function, but otherwise I think having them separate makes reading and maintenance easier.

Reply:
Well, the tests are right in front of my face so I should have just read them. Doing so now, I do think that I would try to summarize them via `@pytest.mark.parametrize`. There seem to be just a couple differences that could be passed in to a generalized test-function body as parameters.